### PR TITLE
[Fonts API] removing files and files loading no longer needed

### LINKED ISSUE #51820: [Fonts API] Remove all files once Font Library is merged
Related to https://github.com/WordPress/gutenberg/issues/51769 and https://github.com/WordPress/gutenberg/issues/51819

Part of https://github.com/WordPress/gutenberg/issues/41479.

Dependent upon https://github.com/WordPress/gutenberg/pull/51770/, https://github.com/WordPress/gutenberg/issues/51819, and https://github.com/WordPress/gutenberg/pull/50927 being merged into `trunk`.

## What problem does this address?

The Fonts API will be dead code once the Font Library is merged into `trunk`, as it will not load into memory. 

## What is your proposed solution?

Once Font Library, Font Face, and https://github.com/WordPress/gutenberg/issues/51819 are all merged into `trunk`, then:

1. Delete the entire `/lib/experimental/fonts-api/` folder.
2. Delete the `require` lines for each of those files in `lib/load.php`.

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
Removing source files around the deprecated Fonts API they are not longer needed.
See: https://github.com/WordPress/gutenberg/issues/51820

## Why?
Fixes: https://github.com/WordPress/gutenberg/issues/51820

## How?
Removing the files and the conditional loading of them.

## Testing Instructions
- Run unit tests

### CODE REVIEW

**Comment by hellofromtonya** [comment]
This PR not only removes the deprecated Fonts API but also turns on the Font Library by default. Will this be a problem (thinking of Jetpack)?

**Comment by hellofromtonya** [comment]
>This PR not only removes the deprecated Fonts API but also turns on the Font Library by default. Will this be a problem (thinking of Jetpack)?

Had a conversation with @matiasbenedetto. Jetpack is now using the Font Library by default, i.e. not the Fonts API. Should be okay then.

**Review by hellofromtonya — APPROVED** [review]
LGTM 👍  (And woohoo for this old, deprecated code to finally be gone.)

**Comment by matiasbenedetto** [comment]
@hellofromtonya I needed to add an additional re-declaration guard on `load.php` :  https://github.com/WordPress/gutenberg/pull/57972/commits/4e24308247399b68c21b4b6075a2c3d7717b7849
I found that was needed because the [CI unit tests were failing](https://github.com/WordPress/gutenberg/pull/57972#issuecomment-1899131806) on PHP 7.0 with this message:
```
Fatal error: Cannot declare class WP_Font_Face, because the name is already in use in /var/www/html/wp-content/plugins/gutenberg/lib/compat/wordpress-6.4/fonts/font-face/class-wp-font-face.php on line 21
```

Could you check this again, please?

**Comment by hellofromtonya** [comment]
Whoops, also need to remove the Fonts API tests. This whole folder can be removed https://github.com/WordPress/gutenberg/tree/trunk/phpunit/tests/fonts-api.

File: lib/load.php [hunk 1]
```diff
@@ -139,78 +139,43 @@ function gutenberg_is_experiment_enabled( $name ) {
...
+require __DIR__ . '/experimental/fonts/font-library/class-wp-font-library.php';
+require __DIR__ . '/experimental/fonts/font-library/class-wp-font-family-utils.php';
+require __DIR__ . '/experimental/fonts/font-library/class-wp-font-family.php';
+require __DIR__ . '/experimental/fonts/font-library/class-wp-rest-font-families-controller.php';
+require __DIR__ . '/experimental/fonts/font-library/class-wp-rest-font-collections-controller.php';
+require __DIR__ . '/experimental/fonts/font-library/class-wp-rest-autosave-font-families-controller.php';
+require __DIR__ . '/experimental/fonts/font-library/font-library.php';
+
+// Load the Font Face and Font Face Resolver, if not already loaded by WordPress Core.
+if ( ! class_exists( 'WP_Font_Face' ) ) {
```
Comment (hellofromtonya) [inline lib/load.php:153]: @matiasbenedetto this check needs to stay as Core loads these files. If the website has < WP 6.4.0, then the files are loaded.

**Comment by hellofromtonya** [comment]
All tests are failing (not just fonts) for:
```
Indirect modification of overloaded property WP_Block_Type::$variations has no effect
```

which was fixed a little bit ago by https://github.com/WordPress/gutenberg/commit/b2c2a085e30a50f51414f819220c9ff974cdd328. This PR will need a rebase to pull the changes into it.

**Review by hellofromtonya — APPROVED** [review]
Re-reviewing. All of the Fonts API files including the tests are now removed.

This PR is ready for merge into `trunk`. Thanks @matiasbenedetto ⭐

**Comment by matiasbenedetto** [comment]
Thanks for the review and the fixes :star2:

**Pull request merged**